compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -1176,6 +1179,12 @@ public void visit(BLangFunction funcNode) {
         BType currentReturnType = this.forceCastReturnType;
         this.forceCastReturnType = null;
         funcNode.body = rewrite(funcNode.body, funcEnv);
+        if (serviceStarted) {
```

Comment:
Lets move this to transactionDesugar or rename it to TransactionCoordinatorService.

Reply:
This approach was modified via https://github.com/ballerina-platform/ballerina-lang/pull/27477